assets/blocks/quiz/quiz-block/quiz-settings.js

```diff
@@ -17,18 +17,20 @@ import { __ } from '@wordpress/i18n';
  */
 import NumberControl from '../../editor-components/number-control';
 import { isQuestionEmpty } from '../data';
+import PaginationSettings from './pagination-settings';
 
 /**
  * Quiz settings.
  *
- * @param {Object}   props                    Block props.
- * @param {Object}   props.attributes         Block attributes
- * @param {Object}   props.attributes.options Current setting options.
- * @param {Function} props.setAttributes      Set attributes function.
- * @param {string}   props.clientId           Block ID.
+ * @param {Object}   props                               Block props.
+ * @param {Object}   props.attributes                    Block attributes
+ * @param {Object}   props.attributes.options            Current setting options.
+ * @param {Function} props.setAttributes                 Set attributes function.
+ * @param {string}   props.clientId                      Block ID.
+ * @param {Object}   props.attributes.paginationSettings Pagination settings.
  */
 const QuizSettings = ( {
-	attributes: { options = {} },
+	attributes: { options = {}, paginationSettings = {} },
```

Comment:
I'm not sure it makes sense to have this as a separate top-level attribute on the block and in the API. Just putting it under `options.pagination` would simplify the code in multiple places.

Reply:
I was a bit on the edge for this one. I kind of like the separation of the settings but indeed it simplifies the code. I have done the change in 0d17faa2ae43e08d1457bf4f252425652091b153  as part of the other request to avoid conflicts.